### PR TITLE
feature/add_class_function

### PR DESCRIPTION
**Pull request by RaphaLK**
<h2> Overview </h2>

<p> So, in this PR we did a handful of tasks such as:

- Added ability to add classes to the four year plan.
- Fixed a handful of CSS issues regarding height/width changes.
- Moved interfaces from the top of the files to a singular file under ../utils.
- Moved list of courses to a separate .ts file.

</p>

### CODE REVIEW

File: src/components/AddClass/AddClassTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  // This function allows for arrow keys to be used to go up and down input boxes
+  const handleKeyDown = (
+    e: React.KeyboardEvent<HTMLInputElement>,
+    index: number
+  ) => {
+    if (e.key === "ArrowDown" && index < inputRefs.length - 1)
+      inputRefs[index + 1]?.current?.focus(); // Move to the next input
+    if (e.key === "ArrowUp" && index > 0)
+      inputRefs[index - 1]?.current?.focus(); // Move to the previous input
+    // TODO: Add a "Submit" button or allow the user to click outside the popup to add classes. Right now, the "Enter" key works.
```
Comment (RaphaLK) [inline src/components/AddClass/AddClassTemplate.tsx:44]: I think this should be a separate from our MVP, what do we all think?

File: src/app/page.tsx [hunk 2]
```diff
@@ -1,37 +1,77 @@
-import AddClass from "@/components/AddClass";
+"use client"; // "The component gets prerendered with SSR or ISR/SSG if possible on the server. The html is send to the client and the javascript is send too. So it gets hydrated on the client and is interactive"
```
Comment (RaphaLK) [inline src/app/page.tsx:1]: This is just to enable hooks

**Review by kayleighpvu — APPROVED** [review]

File: src/components/AddClass/AddClassTemplate.tsx (hunk 1 above)
Comment (walkerselby) [reply src/components/AddClass/AddClassTemplate.tsx:44]: What specifically do you think should be separate? The navigation via keys? If so, yeah, I do not think that is essential.

**Review by walkerselby — APPROVED** [review]

File: src/components/AddClass/AddClassTemplate.tsx (hunk 1 above)
Comment (RaphaLK) [reply src/components/AddClass/AddClassTemplate.tsx:44]: I meant that TODO snippet, right now I have it set to pressing "Enter" to create a class. The separate target could be to add a submit button, or the ability to click "off" the popup to add the class. The navigation via keys is already implemented!

**Pull request merged**